tests/test_uncrustify/test.py

```diff
@@ -140,6 +140,8 @@ def run(self, args):
             if has_diff and self.test_xfail:
                 if args.xdiff:
                     self._diff(_expected, _result)
+                    if not args.show_all:
```

Comment:
This still isn't *quite* consistent... for a normal test, we print the status, then the diff, but here we print the diff, then the status. Maybe normal tests should flip? (I.e. swap lines 133-134 and 135-136...)

Otherwise, in this file we should "always" print the status (i.e. don't check `args.show_all`) and instead in `run_tests`, don't print the statuf if `args.xdiff`.

Reply:
I don't know anymore how we print stuff for a normal test. I think first we need a test framework for the test framework before changing anything here.

I feel like we've tweaked the output 1000 times back and forth... "use these args, no, let's flip them back".  I'm ok in changing them but I don't feel like we're going anywhere with this. We're talking about rare use-cases, combination of flags that almost no one will use.

I think uncrustify could be better off with this PR landed than with a perfect consistent console output handling all combinations of all possible args.